index.js

```diff
@@ -1,184 +1,249 @@
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
-
-
-
+// console.log("%cQuestion 3", "color: red")
+ 
+function drinkWater(current_thirst_level){
+ console.log("Man, I sure am thirsty.")
+ current_thirst_level -=1
+ console.log("Ahh that hits the spot.")
+ return current_thirst_level
+}
+ 
+console.log(drinkWater(12))
+ 
+ 
+ 
+ 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
+// console.log("%cQuestion 4", "color: red")
+ 
+function sameSameButDifferent(num, maybeNum){
+ if (num === maybeNum){
+   return "same same"
+   }
+ else if (num == maybeNum){
+   return "same same (but different)"
+   }
+ else {
+   return "different"
+   }
+ }
+ 
+ 
+ 
+console.log(sameSameButDifferent(5, 5))
 // // => "same same"
-
-// console.log(sameSameButDifferent(123, "123")) 
+ 
+console.log(sameSameButDifferent(123, "123"))
 // // => "same same (but different)"
-
-// console.log(sameSameButDifferent(5, 7)) 
+ 
+console.log(sameSameButDifferent(5, 7))
 // // => "different"
-
-// console.log(sameSameButDifferent(123, "122")) 
+ 
+console.log(sameSameButDifferent(123, "122"))
 // // => "different"
-// console.log("%c----------", "color: red") 
-
-
-
+// console.log("%c----------", "color: red")
+ 
+ 
+ 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
+ 
+ 
+// console.log("%c----------", "color: red")
+ 
+function updateGrade(student, updatedGrade){
+ student.grade = updatedGrade
+}
+ 
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+updateGrade(student1, 67)
 // console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
-
-
-
+ 
+ 
+ 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+ 
+const users = [
+ {
+   name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+ },
+ {
+   name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+ }
+]
+ 
 // // => "Duane"
 // => "Cell: 555-123-4567"
 // => "Office: 555-456-7890"
 // => "Liza"
 // => "Cell: 555-234-5678"
 // => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
-
-
+// console.log("%c----------", "color: red")
+ 
+function printNameAndPhones(users){
+for (let i = 0 < users.length; i++) {
+  console.log(user.name)
+  console.log(user.phones)
+}
+}
+// Uncaught SyntaxError: Unexpected token ')' -- can't figure this out! UGH.
+ 
+ 
+console.log(printNameAndPhones(users))
```

Comment:
Nice attempt. if we are going to use a for loop here then we do not have access to the variable `user` that we are trying to use in your for loop. Instead, we would have to use the full array `users` and the current index `i` to find the individual user.

for (let i = 0 < users.length; i++) {
  console.log(users[i].name)
  console.log(users[i].phones.cell)
  console.log(users[i].phones.office)
}

the other way we could do this would be by using a `forEach` method in which case we would define a `user` variable to use.

function printNameAndPhones(users){
  users.forEach (function(user) {
    console.log(user.name)
    console.log(user.phones.cell)
    console.log(user.phones.office)
  })
}